### PR TITLE
[ADD] module to manage analytic on purchase order

### PR DESCRIPTION
**Pull request by laetitia-gangloff**
This module allow user to set analytic account on purchase order instead of purchase order line.
When analytic account is set on purchase order, all lines of this purchase order will be on this analytic account.

### CODE REVIEW

File: purchase_analytic/models/purchase.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    project_id = fields.Many2one(
+        compute='_compute_project_id',
+        inverse='_inverse_project_id',
+        comodel_name='account.analytic.account',
+        string='Contract / Analytic', readonly=True,
+        states={'draft': [('readonly', False)]},
+        store=True,
+        help="The analytic account related to a sales order.")
+
+    @api.one
```
Comment (andhit-r) [inline purchase_analytic/models/purchase.py]: @api.one deprecated

File: purchase_analytic/models/purchase.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        al = self.project_id2
+        if self.order_line:
+            al = (self.order_line[0].account_analytic_id) or False
+            for ol in self.order_line:
+                if ol.account_analytic_id != al:
+                    al = False
+                    break
+        self.project_id = al
+
+    @api.one
```
Comment (andhit-r) [inline purchase_analytic/models/purchase.py]: @api.one deprecated

**Comment by andhit-r** [comment]
@laetitia-gangloff No unittest for onchange?

**Comment by laetitia-gangloff** [comment]
@andhit-r  thank for your review

There is a test on onchange on purchase order line : https://github.com/acsone/account-analytic/blob/8.0-purchase_analytic-lga/purchase_analytic/tests/test_purchase_analytic.py#L24
And one on project_id change : https://github.com/acsone/account-analytic/blob/8.0-purchase_analytic-lga/purchase_analytic/tests/test_purchase_analytic.py#L50

Is it not sufficient ?

**Comment by andhit-r** [comment]
@laetitia-gangloff sorry i guess i missed it. Other than @api.one usage :+1:

**Comment by laetitia-gangloff** [comment]
Thank you !

I update @api.one in @api.multi

**Comment by yvaucher** [comment]
@laetitia-gangloff What about purchase_analytic_global module which existed in 7.0 ?

https://github.com/OCA/purchase-workflow/tree/7.0/purchase_analytic_global

Which I ported to 9.0: https://github.com/OCA/purchase-workflow/pull/228 (just found out this PR as it was not in the same repository)

**Comment by laetitia-gangloff** [comment]
I miss it :(

**Comment by feketemihai** [comment]
@laetitia-gangloff I think the design is quite complicated, if you want to ease up, why not leave one single field for analytic, add an onchange on it (to change all lines with that value), and the field in context of purchase order lines, if you want to change it for some line, you can still do it...but having an inverse from purchase order lines to purchase order it doesn't make any sense..

**Comment by AaronHForgeFlow** [comment]
Maybe an onchange method would be enough in this case. I'm not against of refactoring this module but this is working fine from a purely functional perspective.

**Comment by AaronHForgeFlow** [comment]
Closing due to inactivity. Please reopen it if you plan to finish this. Regards.

**Pull request closed without merging**